Panic in textinput widget with non-latin1 symbols

OS: confirmed on Windows and Linux
version: `github.com/ebitenui/ebitenui v0.6.3-0.20250515233136-de5d22738a52`

```
panic: runtime error: slice bounds out of range [:35] with capacity 32

goroutine 14 [running]:
github.com/ebitenui/ebitenui/widget.(*TextInput).drawTextAndCaret(0xc002a00300, 0xc00383a300)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/textinput.go:734 +0x67b
github.com/ebitenui/ebitenui/widget.(*TextInput).renderTextAndCaret.func1(0xc00383a380?)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/textinput.go:710 +0x1d
github.com/ebitenui/ebitenui/image.(*MaskedRenderBuffer).Draw(0xc00388e510, 0xc002300b00, 0xc0000356e8, 0xc0000356d8)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/image/buffer.go:59 +0xff
github.com/ebitenui/ebitenui/widget.(*TextInput).renderTextAndCaret(0x67408a?, 0xc002300b00?)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/textinput.go:708 +0x46
github.com/ebitenui/ebitenui/widget.(*TextInput).Render(0xc002a00300, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/textinput.go:352 +0x45
github.com/ebitenui/ebitenui/widget.(*Container).Render(0xc001fdc2a0, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/container.go:261 +0x13d
github.com/ebitenui/ebitenui/widget.(*Container).Render(0xc001fdc690, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/container.go:261 +0x13d
github.com/ebitenui/ebitenui/widget.(*Container).Render(0xc001fdc7e0, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/container.go:261 +0x13d
github.com/ebitenui/ebitenui/widget.(*Container).Render(0xc001fdcbd0, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/container.go:261 +0x13d
github.com/ebitenui/ebitenui/widget.(*Window).Render(0xc0010a4c60, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/widget/window.go:360 +0x673
github.com/ebitenui/ebitenui.(*UI).render(0xc002157880, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/ui.go:154 +0xc5
github.com/ebitenui/ebitenui.(*UI).Draw(0xc002157880, 0xc002300b00)
        github.com/ebitenui/ebitenui@v0.6.3-0.20250515233136-de5d22738a52/ui.go:109 +0x106
```

If we increase the text length, the mismatch gets bigger:
```
panic: runtime error: slice bounds out of range [:68] with capacity 36
```

I would assume a mishandled rune-vs-byte / utf8 length in `cursorPosition` handling somewhere. 

The offending line assumes that `cursorPosition` is a rune position, but in fact that's not always the case if symbols are not latin1-encoded:
```
sub := string([]rune(inputStr)[:t.cursorPosition])
```
I can see at least some places like this:
```
curIdx = len(t.inputText)

// and then
t.cursorPosition = curIdx
```
Where `inputText` is `string`, so `len(t.inputText)` would give something line 40 for a 20-rune string with 2-byte encoded symbols.

--- a/widget/textinput.go
+++ b/widget/textinput.go
@@ -558,7 +558,7 @@ func (t *TextInput) idleState(newKeyOrCommand bool) textInputState {
 			if y < tr.Min.Y {
 				curIdx = 0
 			} else {
-				curIdx = len(t.inputText)
+				curIdx = len([]rune(t.inputText))
 			}
 		}
 		textSize := tr.Dx() - fontAdvance(t.inputText, t.computedParams.Face)

